source/AI.cpp

```diff
@@ -463,13 +463,14 @@ void AI::Step(const PlayerInfo &player)
 			
 			command |= AutoFire(*it);
 		}
-		if(isPresent && personality.Harvests() && DoHarvesting(*it, command))
+		if(isPresent && (personality.Harvests() || (it->IsYours() && Preferences::Has("Escorts harvest flotsam")))
+				&& DoHarvesting(*it, command))
 		{
 			it->SetCommands(command);
 			continue;
 		}
-		if(isPresent && personality.IsMining() && !target
-				&& it->Cargo().Free() >= 5 && ++miningTime[&*it] < 3600 && ++minerCount < 9)
+		if(isPresent && (personality.IsMining() || (it->IsYours() && Preferences::Has("Escorts target asteroids")))
+				&& !target && it->Cargo().Free() >= 5 && ++miningTime[&*it] < 3600 && ++minerCount < 9)
```

Comment:
I'd say player commands should be considered more important then mining personality. Otherwise, your fleet would consist of insubordinate greedlings who ignore your orders to, you know, hold defensible position (ofc you need them to move over there first), for example. _Including_ gather. When player uses gather command, usually it's for a reason (someone's attacking the flagship, for example). The last thing you want at this point is fleet ignoring your commands. Mining personality should kick in when your ships would be otherwise idle (as in, no active orders).

Alternatively, you could implement so that while "gather" is active, they still mine, but only at certain radius (immediate vicinity) around flagship, and do their best to not stray too far away from her).